src/server.py

```diff
@@ -116,33 +121,66 @@ def parse_req(request):
     return request_decon[1]
 
 
-def response_decision(full_mes):
+def response_decision(request):
     """Based on the raised error, return an appropriate http-exception
      and return URI = 'None'. Return an ok-response and URI for a
      good request. Response = byte-string, uri = unicode-string."""
     uri = u'None'
     try:
-        parse_req(full_mes)
+        parse_req(request)
     except NotImplementedError:
         response = HTTPException(u'405', u'Method Not Allowed',
-                                 u'The server supports HTTP/1.1 only.')\
-            .response_msg()
+                                 u'The server supports HTTP/1.1 only.')
+        response = response.response_msg()
     except TypeError:
         response = HTTPException(u'505', u'HTTP Version Not supported',
-                                 u'The server supports HTTP/1.1 only.')\
-            .response_msg()
+                                 u'The server supports HTTP/1.1 only.')
+        response = response.response_msg()
     except NameError:
         response = HTTPException(u'400', u'Bad Request',
-                                 u'No <host> in headers.')\
-            .response_msg()
+                                 u'No <host> in headers.')
+        response = response.response_msg()
     except:
         response = HTTPException(u'500', u'Internal Server Error',
-                                 u'Something went wrong.')\
-            .response_msg()
+                                 u'Something went wrong.')
+        response = response.response_msg()
+    else:
+        uri = parse_req(request)
+        response = generate_response(uri)
+    return response
```

Comment:
Okay, so now the value that will be returned from this function is goiing to be a response, no matter what.  But if there was a problem, the response will be an error response.  If there was no problem it will be an ok response.  

So how do you tell the difference between one and another when it comes time to test?